src/ui/public/doc_table/components/table_row.js

```diff
@@ -92,18 +102,30 @@ module.directive('kbnTableRow', function ($compile) {
           openRowHtml
         ];
 
+        const mapping = indexPattern.fields.byName;
         if (indexPattern.timeFieldName) {
           newHtmls.push(cellTemplate({
             timefield: true,
-            formatted: _displayField(row, indexPattern.timeFieldName)
+            formatted: _displayField(row, indexPattern.timeFieldName),
+            filterable: mapping[indexPattern.timeFieldName].filterable,
+            column: indexPattern.timeFieldName
           }));
         }
 
         $scope.columns.forEach(function (column) {
+          const flattenedRow = indexPattern.flattenHit(row);
+          const isFilterable =
+            flattenedRow[column] === undefined
+            ? false
+            : !mapping[column]
+            ? false
+            : mapping[column].filterable;
```

Comment:
So it turns out we have an eslint rule against nested ternaries:

```
/var/lib/jenkins/workspace/elastic+kibana+pull-request+multijob-intake/src/ui/public/doc_table/components/table_row.js
  117:13  error  Do not nest ternary expressions  no-nested-ternary
```

@jimmyjones2 would you mind implementing the function @cjcenizal suggested above to replace the ternary? Sorry for the back and forth, this should be the last change assuming tests pass.